source/to_cpp1.h

```diff
@@ -2772,6 +2797,89 @@ class cppfront
         return is_pointer_declaration(decl, deref_cnt, addr_cnt);
     }
 
+
+    auto source_order_name_lookup(unqualified_id_node const& id)
```

Comment:
[NSDMI]: https://quuxplusone.github.io/blog/2019/08/02/the-tough-guide-to-cpp-acronyms/#nsdmi

I've been putting off dealing with type scope name lookup,
which is more complicated than just source order name lookup:
<https://en.cppreference.com/w/cpp/language/unqualified_lookup#Class_definition>.
Using source order name lookup is wrong
because it doesn't consider members declared later
in member function bodies and [NSDMI][]s.

Reply:
[NSDMI]: https://quuxplusone.github.io/blog/2019/08/02/the-tough-guide-to-cpp-acronyms/#nsdmi

This is how I've been thinking of dealing with this.
In Phase 1 "Cpp2 type declarations", cache the member names in the type.
When emitting code within a [NSDMI][] or member function body,
use a lookup algorithm that considers the cached member names.